docs/IR.md

```diff
@@ -55,12 +55,13 @@ The following describes the metadata properties of a model graph:
 
 #### Names Within a Graph
 
-Names are organized into separate namespaces, and must be unique within a namespace. The namespaces include the following:
- - `Node`: names that identify specific nodes in the graph, but not necessarily any particular input or output of the node.
- - `Graph`: names that identify graphs in the protobuf.
+Names are organized into separate namespaces. The namespaces include the following:
+ - `Node`: names that identify specific nodes in the graph, but not necessarily any particular input or output of the node. Each node name must be unique.
```

Comment:
We should add scope for each type of names. For example, Node name must be unique in the whole model.

Reply:
Yep, we should discuss it - or maybe make it graph-level unique? @jamesr66a probably has a view on it given the control flow considerations

Btw, shall it be "optional"?